sale_automatic_workflow_periodicity/models/automatic_workflow_job.py

```diff
@@ -0,0 +1,42 @@
+# Copyright 2023 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+from datetime import datetime, timedelta
+
+from odoo import api, models
+
+
+class AutomaticWorkflowJob(models.Model):
+    _inherit = "automatic.workflow.job"
+
+    def run_with_workflow(self, sale_workflow):
+        res = super().run_with_workflow(sale_workflow)
+        if sale_workflow.periodicity:
+            sale_workflow.next_execution = datetime.now() + timedelta(
+                seconds=sale_workflow.periodicity
+            )
```

Comment:
Better to use Odoo API:
```suggestion
            now = fields.Datetime.now()
            sale_workflow.next_execution = fields.Datetime.add(now, seconds=sale_workflow.periodicity)
```

Reply:
Any reason why ? 
As `fields` is not yet imported.

Reply:
Because we are in Odoo context